pkg/services/provisioning/dashboards/types.go

```diff
@@ -91,13 +97,19 @@ func mapV0ToDashboardAsConfig(v0 []*DashboardsAsConfigV0) []*DashboardsAsConfig
 		})
 	}
 
-	return r
+	return r, nil
 }
 
-func (dc *DashboardAsConfigV1) mapToDashboardAsConfig() []*DashboardsAsConfig {
+func (dc *DashboardAsConfigV1) mapToDashboardAsConfig() ([]*DashboardsAsConfig, error) {
 	var r []*DashboardsAsConfig
+	seen := make(map[string]bool)
 
 	for _, v := range dc.Providers {
+		if _, ok := seen[v.Name.Value()]; ok {
+			return nil, fmt.Errorf("name %s is not unique", v.Name.Value())
```

Comment:
To make sure the error comes out as unambiguous (e.g. in case the name has spaces in it), I'd prefer quoting the name and also clarify it's that of a dashboard:
```suggestion
			return nil, fmt.Errorf("dashboard name %q is not unique", v.Name.Value())
```